modules/serverless-rn-1-24-0.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies
+//
+// * /serverless/serverless-release-notes.adoc
+
+:_content-type: REFERENCE
+[id="serverless-rn-1-24-0_{context}"]
+= Release notes for Red Hat {ServerlessProductName} 1.24.0
+
+{ServerlessProductName} 1.24.0 is now available. New features, changes, and known issues that pertain to {ServerlessProductName} on {product-title} are included in this topic.
+
+[id="new-features-1.24.0_{context}"]
+== New features
+
+* {ServerlessProductName} now uses Knative Serving 1.3.
+* {ServerlessProductName} now uses Knative Eventing 1.3.
+* {ServerlessProductName} now uses Kourier 1.3.
+* {ServerlessProductName} now uses Knative `kn` CLI 1.3.
+* {ServerlessProductName} now uses Knative Kafka 1.3.
+* The `kn func` CLI plug-in now uses `func` 0.24.
+
+* Init containers support for Knative services is now generally available (GA).
+
+ifdef::openshift-enterprise[]
+* You can now use the cost management service with {ServerlessProductName}.
+endif::[]
+
+[id="fixed-issues-1.24.0_{context}"]
+== Fixed issues
+
+* Integrating {ServerlessProductName} with {SMProductName} causes the `net-istio-controller` pod to run out of memory on startup when too many secrets are present on the cluster.
++
+It is now possible to enable secret filtering, which makes `net-istio-controller` to only consider secrets with a `networking.internal.knative.dev/certificate-uid` label, thus no longer requiring an increased memory limit.
```

Comment:
```suggestion
* In previous versions, integrating {ServerlessProductName} with {SMProductName} caused the `net-istio-controller` pod to run out of memory on startup when too many secrets were present on the cluster. It is now possible to enable secret filtering so that `net-istio-controller` only considers secrets with a `networking.internal.knative.dev/certificate-uid` label. This means that the `net-istio-controller` pod no longer requires an increased memory limit.
```

Reply:
@abrennan89 Integrating {ServerlessProductName} with {SMProductName} still causes that, so we shouldn't state that this applies only to previous versions. And I prefer to keep the paragraphs separate here, as they are very distinct.

Reply:
In that case I'd separate them entirely and make the first paragraph a known issue instead

Reply:
@abrennan89 It's not an issue, it's expected that a big number of secrets require a big amount of memory. It's just now we provide a feature of reducing the number of secrets => reducing amount of required memory. The first paragraph provides the reason for having the feature.
@maschmid Please correct me if I'm wrong.

Reply:
Resolving this, per Marek's approve. Feel free to reopen if still not agreeing.

Reply:
Both viewpoints make sense IMO,

you can either view it as a bug that it needs to keep all cluster Secrets in memory, with the secret filtering being a kind of workaround;

or you could view the memory requirements as inherent to how the `net-istio-controller` informer works, with the secret filtering being an additional feature to allow you to choose (higher memory limits for `net-istio-controller`, or secret filtering)

FWIW, according to the state of https://issues.redhat.com/browse/SRVKS-900, engineering considers the addition of secret filtering feature as being the fix for the issue. (there's no open known issue for "`net-istio-controller` needs to keep all the cluster secrets in memory")